### ISSUE TITLE
Global values

### ISSUE THREAD

**Issue by kfrost89** [peterramsing/lost#261]
I have encountered an issue with global values in lost.

Initially, when a global value is set, there seems to be no issues. However, changing or removing a global value will cause a bug. I have only tested this with the flex value.

An example:

@lost flexbox flex;

This changes all to flex. However, if I remove the global, the global value still exists. I simply cannot remove it. In some rare cases, the bug suddenly disappears and flex is removed entirely.

**Comment by peterramsing** [comment]
Hmm. That looks odd. Could you provide some code for us to reference?

**Comment by kfrost89** [comment]
I have done some more digging. Initially, the problem occurred on codepen.io. It seems as if that has been fixed. However, the issue also occurred locally.

I think I have located the issue but it does not relate to global values but media queries instead.

I suggest that I open another issue and this could be closed.

**Comment by dacodekid** [comment]
@peterramsing I'm also facing the same issue. I'm using it with `webpack` with following config

``` javascript
  postcss: [
    require('postcss-import')({ addDependencyTo: this }),
    require('postcss-cssnext'),
    require('lost')
  ],
```

Every time I change the global variable, it works fine. But if I comment them, then I had to restart `webpack` as the old values still exists on hot reload. I tried it both in `stylus & css`. This is my test code.

``` css
@lost flexbox flex;
@lost gutter 100px;  /* Commenting this line won't work */

.data-table {
  lost-center: 80%;
}
```

**Comment by corysimmons** [comment]
I've been seeing similar behavior on CodePen.

@ai @ben-eb Do either of you know what might be wrong with this? Here is the small file where we parse at-rules for global settings: https://github.com/peterramsing/lost/blob/master/lib/lost-at-rule.js

**Comment by ben-eb** [comment]
You've set settings globally so it will keep the value for the next time you use it, regardless of the atrule being there or not.

**Comment by corysimmons** [comment]
Are you saying these JS vars are getting cached or something? Do you happen to know of a solution to this?

**Comment by ben-eb** [comment]
Scope the options to be available for each processing step instead of globally.

``` js
postcss.plugin('', function () {
  return function (css, result) {
    var opts = {};
  };
});
```

**Comment by corysimmons** [comment]
@ben-eb Thanks for the tip. =)

Wait, are we not doing that here? https://github.com/peterramsing/lost/blob/master/lost.js#L27-L35

**Comment by ben-eb** [comment]
Setting options in the function that modifies the CSS means that the variable is only around for as long as that CSS tree is. I'm sure you will be able to reproduce the problem by processing two strings of CSS through the same PostCSS instance.

```
var proc = postcss(lost);

var res1 = proc.process(css).css;
var res2 = proc.process(css2).css;
```

**Comment by peterramsing** [comment]
I'll take a look at this this weekend.

**Comment by peterramsing** [comment]
Fixed by #312
